### ISSUE TITLE
clamav and freshclam errors on container creation

### ISSUE THREAD

**Issue by Nebukad93** [hardware/mailserver#175]
I had this errors in my logs after recreating the mailserver container : 

```
2017-09-27T11:52:51.457585+00:00 xx root: s6-supervise : spawning freshclam process
ERROR: Problem with internal logger (UpdateLogFile = /var/log/clamav/freshclam.log).
ERROR: Can't open /var/log/clamav/freshclam.log in append mode (check permissions!).
2017-09-27T11:52:51.480608+00:00 xx root: s6-supervise : virus database downloaded, spawning clamd process
2017-09-27T11:52:52.458933+00:00 xx root: s6-supervise : spawning freshclam process
ERROR: Problem with internal logger (UpdateLogFile = /var/log/clamav/freshclam.log).
ERROR: Can't open /var/log/clamav/freshclam.log in append mode (check permissions!).
2017-09-27T11:52:52.481151+00:00 xx root: s6-supervise : virus database downloaded, spawning clamd process
2017-09-27T11:53:02.213469+00:00 xx root: s6-supervise : freshclam has restarted too many times (permanent failure)
2017-09-27T11:53:02.215210+00:00 xx root: s6-supervise : clamd has restarted too many times (permanent failure)
```

I had to exec a terminal in the container to change the logfiles ownership from root:root to 101:103 in order to solve this.

**Comment by hardware** [comment]
You have this error everytime ?

I can't reproduce your problem on my end with the last docker hub build : 

```
ls -la /var/log/clamav
total 8
drwxr-xr-x 2 clamav clamav   45 Sep 27 16:02 .
drwxr-xr-x 5 root   root    172 Sep 27 16:03 ..
-rw-r----- 1 clamav adm    2767 Sep 27 16:05 clamav.log
-rw-r----- 1 clamav adm    1538 Sep 27 16:04 freshclam.log
```
```
stat /var/log/clamav/freshclam.log
  File: /var/log/clamav/freshclam.log
  Size: 2072            Blocks: 8          IO Block: 4096   regular file
Device: fe03h/65027d    Inode: 83888545    Links: 1
Access: (0640/-rw-r-----)  Uid: (  101/  clamav)   Gid: (    4/     adm)
Access: 2017-09-27 16:02:16.000000000 +0000
Modify: 2017-09-27 17:04:04.122992490 +0000
Change: 2017-09-27 17:04:04.122992490 +0000
 Birth: 
```
```
grep "User" /etc/clamav/clamd.conf
User clamav
```
```
grep "DatabaseOwner" /etc/clamav/freshclam.conf
DatabaseOwner clamav
```

**Comment by hardware** [comment]
Maybe this file is already used by another process or multiple freshclam processes. Can you give me the result of the following command when this error occurs :

```
lsof +d /var/log/clamav
```

**Comment by Nebukad93** [comment]
I happens when I create the container with the latest image from hub.docker.com. I'm using docker's user namespace security option like before.

I've just done a restart, and the ownership was back to root:root with the container still UP but with no clamav (the used memory is down to 50Mo).

I change back ownership to clamav:clamav or clamav:adm and restart again the container, then It's OK.

I can't run the last command in the container, It says "lsof command not fund".

**Comment by hardware** [comment]
> I can't run the last command in the container, It says "lsof command not fund".

When a command doesn't exist, what do you think you should do ?

Solution :

`apt-get upgrade && apt-get install lsof`

**Comment by Nebukad93** [comment]
Sorry, I didn’t see my last sentence was incomplete :
After install there is no result and no output.

**Comment by hardware** [comment]
Your issue should be fixed with this commit : 79933b307d0c2ab57c3dba4326b8e697fd7e6e25 `1.1-latest`

**Comment by Nebukad93** [comment]
Thanks !
